--- a/xio-openssl.c
+++ b/xio-openssl.c
@@ -1900,5 +1900,5 @@
 			/* binary address format */
-			const unsigned char *data = pName->d.iPAddress->data;
-			size_t len = pName->d.iPAddress->length;
+			const unsigned char *data = ASN1_STRING_get0_data(pName->d.iPAddress);
+			size_t len = ASN1_STRING_length(pName->d.iPAddress);
 			char aBuffer[INET6_ADDRSTRLEN]; 	/* canonical peername */
 			struct in6_addr ip6bin;
